src/main/java/com/github/matidominati/medicalclinic/exception/ApiExceptionHandler.java

```diff
@@ -0,0 +1,19 @@
+package com.github.matidominati.medicalclinic.exception;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.time.ZonedDateTime;
+
+@ControllerAdvice
+public class ApiExceptionHandler {
+    @ExceptionHandler(ApiRequestException.class)
+    public ResponseEntity<Object> handleApiRequestException(ApiRequestException ex) {
+        return ResponseEntity.status(ex.getHttpStatus()).body(ApiErrorResponse.builder()
+                .message(ex.getMessage())
+                .httpStatus(ex.getHttpStatus())
+                        .timestamp(ZonedDateTime.now())
+                .build());
+    }
```

Comment:
Jako, ze wszystkie wyjatki unchecked w Javie dziedzicza po RuntimeException to chcielibysmy napisac handlera wlasnie do wyjatku RuntimeException, ktory nam wlasnie rzuci dla takiego wyjatku status 500 i message jak w tasku.